pandas/core/algorithms.py

```diff
@@ -468,18 +470,16 @@ def _factorize_array(values, na_sentinel: int = -1, size_hint=None, na_value=Non
 
     Returns
     -------
-    labels : ndarray
+    codes : ndarray
```

Comment:
do we necessarily know the dtype?  e.g. could this be `ndarray[int64]`?

Reply:
As it's implemented, codes will always be of type ``ndarray[int64]``. I've got a suspicion taht there are cases where would like it to return arrays of fewer bits, but yes, as it is now, it always is an int64 array.